### PR TITLE
Simplify `report` util even further

### PR DESCRIPTION
**Pull request by Mouvedia**
> Which issue, if any, is this issue related to?

https://github.com/stylelint/stylelint/pull/7762#discussion_r1642584242

> Is there anything in the PR that needs further explanation?

N/A

### CODE REVIEW

File: lib/utils/report.mjs [hunk 1]
```diff
@@ -170,28 +154,18 @@ function isDisabled(ruleName, startLine, disabledRanges) {
...
 
-/** @param {Problem} problem */
-function isFixApplied(problem) {
-	const { end, start, index, endIndex, fix, node, result, ruleName, word } = problem;
-	const { disabledRanges, config = {}, fixersData } = result.stylelint;
+/** @param {Problem & { line: number }} problem */
+function isFixApplied({ fix, line, result: { stylelint }, ruleName }) {
+	const { disabledRanges, config = {}, fixersData } = stylelint;
 
 	if (typeof fix !== 'function') {
```
Comment (ybiquitous) [inline lib/utils/report.mjs:161]: [suggest] How about replacing `typeof` here as well? Other `typeof` in this file can be addressed, too.

```suggestion
	if (!isFn(fix)) {
```

File: lib/utils/report.mjs [hunk 2]
```diff
@@ -16,54 +16,45 @@ import { isNumber } from './validateTypes.mjs';
...
+	const ruleSeverity = isFn(severity) ? severity(...messageArgs) || defaultSeverity : severity;
 
 	// In quiet mode, mere warnings are ignored
-	if (result.stylelint.quiet && ruleSeverity !== 'error') {
-		return;
-	}
-
-	// endIndex is optional
-	const { start } = (node && node.rangeBy({ index, endIndex })) || {};
+	if (quiet && ruleSeverity !== 'error') return;
```
Comment (ybiquitous) [inline lib/utils/report.mjs:32]: [suggest] How about extracting the severity literals to constants (`'error'` and `'warning'`)? Put the constants in `lib/constants.mjs` seems good.

```suggestion
	if (quiet && ruleSeverity !== SEVERITY_ERROR) return;
```

Comment (Mouvedia) [reply lib/utils/report.mjs:32]: There are too matches in the code base. Id prefer if it was done in a separate refactor PR. 
i.e. it should be done for all files at once

Comment (ybiquitous) [reply lib/utils/report.mjs:32]: I'm fine with another PR 👍🏼

**Review by ybiquitous — APPROVED** [review]
Now much simpler. LGTM 👍🏼

File: lib/utils/report.mjs [hunk 3]
```diff
@@ -16,54 +16,45 @@ import { isNumber } from './validateTypes.mjs';
...
 
+	const range = node?.rangeBy({ index, endIndex, word }) ?? {};
```
Comment (romainmenke) [inline lib/utils/report.mjs:34]: @Mouvedia Can you elaborate on this change?
Why is `word` included here?

I am fairly certain that this is a breaking change.

See : https://github.com/postcss/postcss/blob/f9a9868eb3a591cd99d504ac57bf7ee77380856b/lib/node.js#L222-L229

Comment (Mouvedia) [reply lib/utils/report.mjs:34]: #7779

**Pull request merged**